ophyd/ophydobj.py

```diff
@@ -97,6 +97,50 @@ def __init__(self, *, name=None, attr_name='', parent=None, labels=None,
         # Instantiate logger
         self.log = logging.getLogger(base_log + '.' + name)
 
+    def __init_subclass__(cls, version=None, version_of=None, **kwargs):
+        'This is called automatically in Python for all subclasses of OphydObject'
+        super().__init_subclass__(**kwargs)
+
+        if version is None:
+            # Only in the simple case of single inheritance, copy in the parent
+            # version information
+            try:
+                parent_class, = cls.__bases__
+            except ValueError:
+                parent_info = None
+            else:
+                parent_info = getattr(parent_class, '_class_info_', None)
+
+            if parent_info is not None:
+                # This class does not *define* a version, but it may be of
+                # a specific version.
+                # Propagate the version information from the parent
+                cls._class_info_ = parent_info.copy()
```

Comment:
I'll have to remove this portion - but it will further simplify `__init_subclass__`. 

We can still figure out which version an object is by inspecting `cls.mro()._class_info_` and then checking `issubclass(new_obj, cls_version) for ver, cls_version in _class_info_['versions'].items()` (semi-pseudo-code here, of course)